giraffe/src/components/Geo.tsx

```diff
@@ -0,0 +1,263 @@
+// Libraries
+import React, {createRef, PureComponent} from 'react'
+import BingLayer from './geo/bing-maps/Bing'
+import {Map, TileLayer, LayersControl} from 'react-leaflet'
+import Control from 'react-leaflet-control'
+import 'leaflet/dist/leaflet.css'
+
+// Components
+import CircleMarkerLayer from './geo/CircleMarkerLayer'
+import HeatmapLayer from './geo/HeatmapLayer'
+import PointMapLayer from './geo/PointMapLayer'
+import TrackMapLayer from './geo/TrackMapLayer'
+
+// Utils
+import {preprocessData} from './geo/processing/tableProcessing'
+
+// Types
+import {
+  GeoCircleViewLayer,
+  GeoHeatMapViewLayer,
+  GeoLayerConfig,
+  GeoPointMapViewLayer,
+  GeoTrackMapViewLayer,
+  GeoViewLayer,
+} from '../types/geo'
+import {GeoTable} from './geo/processing/GeoTable'
+import {Config, Table} from '../types'
+
+// Constants
+const ZOOM_FRACTION = 8
+
+interface Props extends Partial<GeoLayerConfig> {
+  width: number
+  height: number
+  table: Table
+  stylingConfig: Partial<Config>
+}
+
+interface State {}
+
+class Geo extends PureComponent<Props, State> {
```

Comment:
Can we please use Function Components with Hooks rather than Class Components? While Facebook has no immediate plans to deprecate Class Components, the future of React is Function Components with Hooks.

Please let me know if you need help with this. I would be happy to contribute to this effort.

Reply:
Sure. I started with react components as I saw these were used in the Ghronograf code, later I bumped into function components and I started to use these instead. Didn't know there is such policy/effort to use function components only.

I will start to change this in places where it is going to be easy and I will ask for help when I get into trouble (Thanks for the offer!). 
I have spent a lot of time with the current code (and the Chronograf counter part) so that the map gets re-rendered only when it is needed (for better performance / responsiveness) because chronograf has been triggering false rendering calls into the widget. I am afraid this will get a hit during the conversion ;-)

Reply:
It is done.